src/App.js

```diff
@@ -14,11 +16,33 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    load: false,
+  }
+
+  click = () => {
+    this.setState({
+      load: true,
+    });
+  }
+
+  render() {
+    const { load } = this.state;
+
+    return (
+      <div className="App">
+        {load ? (<GoodList goodList={goodsFromServer} />) : (
```

Comment:
there is no need to use goods list in this component as you don't use it here, import in the component you use them